packages/database/src/changeHandlers/TaskCreationHandler.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+import keyBy from 'lodash.keyby';
+import { ChangeHandler } from './ChangeHandler';
+
+const getAnswerWrapper = (config, answers) => {
+  const answersByQuestionId = keyBy(answers, 'question_id');
+
+  return questionKey => {
+    const questionId = config[questionKey]?.questionId;
+    if (!questionId) {
+      return null;
+    }
+    const answer = answersByQuestionId[questionId];
+    return answer?.text;
+  };
+};
+
+const isPrimaryEntityQuestion = (config, questions) => {
+  const primaryEntityQuestion = questions.find(question => question.type === 'PrimaryEntity');
+  const { questionId } = config['entityId'];
+  return primaryEntityQuestion.id === questionId;
+};
+
+const getSurveyCode = async (models, config) => {
+  const surveyCode = config.surveyCode;
+  const survey = await models.survey.findOne({ code: surveyCode });
+  return survey.id;
+};
+
+const getQuestions = (models, surveyId) => {
+  return models.database.executeSql(
+    `
+        SELECT q.*, ssc.config::json as config
+        FROM question q
+        JOIN survey_screen_component ssc ON ssc.question_id = q.id
+        JOIN survey_screen ss ON ss.id = ssc.screen_id
+        WHERE ss.survey_id = ?;
+      `,
+    [surveyId],
+  );
+};
+
+export class TaskCreationHandler extends ChangeHandler {
+  constructor(models) {
+    super(models, 'task-creation-handler');
+
+    this.changeTranslators = {
+      surveyResponse: change => this.getNewSurveyResponses(change),
+    };
+  }
+
+  /**
+   * @private
+   * Only get the new survey responses that are created, as we only want to mark tasks as completed when a survey response is created, not when it is updated
```

Comment:
Just need to update this comment (which I'm assuming is copied over from the other task change handler)